### PR TITLE
feat(apigee): Add support for NatAddress activation

### PR DESCRIPTION
**Pull request by entertvl**
- fix https://github.com/hashicorp/terraform-provider-google/issues/13914

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
apigee: Add support for NAT Address activation in `google_apigee_nat_address`.
```

### CODE REVIEW

**Comment by entertvl** [comment]
> Is it possible to do what you need to do in the read method?

Resolved using encoder/decoder. Thanks.

PR is ready for review. Could you take a look?

**Comment by entertvl** [comment]
What reason of failed to terraform-google-conversion-test-integration step?
I don't know how the next.

**Comment by trodge** [comment]
My apologies, that test has been a bit flakey since the 6.0.0 changes. I'm trying re-running the tests. If that fails, rebasing this PR may fix it.

**Comment by entertvl** [comment]
Rebased and checked that the re-run tests passed locally.

```
--- PASS: TestAccApigeeNatAddress_apigeeNatAddressBasicTestExample (2463.60s)
--- PASS: TestAccApigeeNatAddress_apigeeNatAddressUpdateTest (2579.36s)
```

File: mmv1/templates/terraform/constants/apigee_nat_address.go.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		if state == "CREATING" {
+			return retry.RetryableError(fmt.Errorf("NatAddress %q has state %q.", id, state))
+		} else if state == "RESERVED" {
+			log.Printf("[DEBUG] NatAddress %q has state %q.", id, state)
+			return nil
+		} else {
+			return retry.NonRetryableError(fmt.Errorf("NatAddress %q has state %q.", id, state))
+		}
+	})
+}
```
Comment (trodge) [inline mmv1/templates/terraform/constants/apigee_nat_address.go.erb:20]: Here's a fix for the TGC build/unit test that's failing.
```suggestion
<% unless compiler == "terraformgoogleconversion-codegen" -%>
// waitForNatAddressReady waits for an NatAddress to leave the
// "CREATING" state and become "RESERVED", to indicate that it's ready.
func waitForNatAddressReserved(d *schema.ResourceData, config *transport_tpg.Config, timeout time.Duration) error {
	return retry.Retry(timeout, func() *retry.RetryError {
		if err := resourceApigeeNatAddressRead(d, config); err != nil {
			return retry.NonRetryableError(err)
		}

		id := d.Id()
		state := d.Get("state").(string)
		if state == "CREATING" {
			return retry.RetryableError(fmt.Errorf("NatAddress %q has state %q.", id, state))
		} else if state == "RESERVED" {
			log.Printf("[DEBUG] NatAddress %q has state %q.", id, state)
			return nil
		} else {
			return retry.NonRetryableError(fmt.Errorf("NatAddress %q has state %q.", id, state))
		}
	})
}
<% end -%>
```

**Review by trodge — CHANGES_REQUESTED** [review]

**Review by trodge — APPROVED** [review]

**Pull request merged**